### ISSUE TITLE
Camera in the editor stutters even in an empty scene

### ISSUE THREAD

**Issue by nxrighthere** [OurMachinery/themachinery-public#64]
You can see what exactly happens in this video: [2020-01-12 21-41-55.zip](https://github.com/OurMachinery/themachinery-public/files/4050847/2020-01-12.21-41-55.zip)

My specs: AMD Ryzen 5 1400, GTX 1660, 16 Gb DDR4 RAM, Windows 10 Build 1909.

**Comment by niklas-ourmachinery** [comment]
Yes, this should not normally happen.

Can you use the profiler to identify the source of the stutters?

* Open a profiler view (View > Profiler)
* Fly around with the camera to produce stutters.
* Stop the profiling session.
* Zoom out and scroll back (WASD) in the profiler view until you see an entry that is much bigger than the other  ones..

That big entry should correspond to the stutter... with some luck we can tell in which system it's happening.

**Comment by nxrighthere** [comment]
Hey, so here's the trace: [trace.zip](https://github.com/OurMachinery/themachinery-public/files/4105925/trace.zip)

So, from the profiler perspective I think this is how the stutter looks like:
![blank](https://user-images.githubusercontent.com/8279330/73030458-80149500-3e5b-11ea-9364-831d62dce51e.PNG)
Pretty much just a blank space, but I'm not sure that this is it, because the profiler is capturing data so fast I can't even catch with my eye when exactly this happens and how it looks like.
![blank2](https://user-images.githubusercontent.com/8279330/73030602-d4b81000-3e5b-11ea-9d56-dc71b1ed1b52.PNG)

**Comment by niklas-ourmachinery** [comment]
Note that you can pause the profiler by pressing the running MB number, that makes it a bit easier to zoom in and find the issue.

**Comment by nxrighthere** [comment]
Yes, I think it would be more convenient if we will be able to control the update frequency of the profiler to slow it down a bit. For example, the profiler in Unity is updating once per second or something like that.

**Comment by nxrighthere** [comment]
Also, when I resume capturing, it automatically reset the zoom level. 😢

**Comment by niklas-ourmachinery** [comment]
I think it is better that it updates as fast as it can so you can see what is going on... otherwise you won't see fast events happening.

To find issues like this what I usually do is record for a bit, making sure I capture a couple of stalls, then pause the profiler and finally scroll back into the history of the profiler to see if I can find the stalls.

Anyway, I loaded your saved profiler data and it's pretty clear it is the `render_task` that is stalling, sometimes taking as much as 100 ms:

![image](https://user-images.githubusercontent.com/26548514/73031631-8a01ae00-3df1-11ea-9ba6-68f4a49d471a.png)

Do you have the latest GPU drivers?

@tobias-ourmachinery do you have any idea of what might be going on?

**Comment by nxrighthere** [comment]
> Anyway, I loaded your saved profiler data and it's pretty clear it is the render_task that is stalling, sometimes taking as much as 100 ms:

Ah, good to know, thanks. Just not sure how suspicions samples are looks like since not very familiar with the engine at the moment. 

> Do you have the latest GPU drivers?

Yes, 441.87 version. With old drivers, these stutters are happening too.

**Comment by nxrighthere** [comment]
@niklas-ourmachinery By the way, what application you are using to load json traces?

**Comment by niklas-ourmachinery** [comment]
It looks very suspicious... the `render_task` should not be taking that long, but that is more @tobias-ourmachinery 's expertise than mine, so I can't get what's happening.

You can load the traces into Chrome. Chrome > More Tools > Developer Tools, and then click Performance and the arrow in the top left corner to load a profile.

We should really support loading them into The Machinery though...

**Comment by nxrighthere** [comment]
Thanks! Just loaded it into the new Microsoft Edge and works great:
![profiler](https://user-images.githubusercontent.com/8279330/73032461-f962b680-3e60-11ea-8eb9-42ea80754442.PNG)
[Speedscope](https://github.com/jlfwong/speedscope) doesn't work, unfortunately, while it states that Chrome profiler is [supported](https://github.com/jlfwong/speedscope/wiki/Importing-from-Chrome). 🤔

**Comment by nxrighthere** [comment]
This issue is still a thing in a freshly released beta. I've tried three different machines with GTX 950, 1050, and 1660, using the latest GPU drivers.

**Comment by nxrighthere** [comment]
Can it be a [similar](https://forum.unity.com/threads/time-deltatime-not-constant-vsync-camerafollow-and-jitter.430339/page-4#post-4173379) problem?

**Comment by nxrighthere** [comment]
Still a thing on 2020.5 version.

**Comment by nxrighthere** [comment]
2020.7 still there.
